Copy Jetpack Compose Material3 1.5.0-alpha10 and Adaptive 1.3.0-alpha05

eymar opened the pull request:
| GroupId | ReleaseVersion | ReleaseSHA | ReleaseBuildId | ReleaseDate |
| --- | --- | --- | --- | --- |
| androidx.compose.material3 | 1.5.0-alpha10 | deb96499dfe95073f5c1215c1287787683cb1e92 | 14504571 | 12/3/2025 |
| androidx.compose.material3.adaptive | 1.3.0-alpha05 | deb96499dfe95073f5c1215c1287787683cb1e92 | 14504571 | 12/3/2025 |

 Fixes [CMP-9378](https://youtrack.jetbrains.com/issue/CMP-9378)
 
 Follow up tasks:
 - https://youtrack.jetbrains.com/issue/CMP-9305

## Testing
N/A

## Release Notes
N/A

eymar committed:
Copy Compose Material3 and Adaptive from dfc34b29f7e

eymar committed:
update material3 api dump

eymar commented:
There are 2 new modules: 
- androidx.compose.material3:material3-xr
- androidx.compose.material3:material3-xr-adaptive

But there are no artefacts for them because they are `INTERNAL_LIBRARY_WITH_API_TASKS`.
I keep these new modules disabled in our fork for now.

eymar committed:
update material3 and adaptive redirection versions

igordmn commented on compose/material3/material3-xr-adaptive/api/current.txt:
```diff
@@ -0,0 +1,13 @@
+// Signature format: 4.0
```
> Copy Compose Material3 and Adaptive from deb9649

The commit points to the `androidx-main` branch, but should point to the commit from `integration`, from which the copy script applied.

Is the title wrong, or is the commit from which it is copied wrong?

igordmn replied:
I see that the branch is `...dfc34b29f7e/`, so probably it is just title? Was it changed manually after applying the script?

eymar replied:
yes. It's a title.

I did "Undo" the original commit (generated by the script) to remove changes in build.gradle and stubs (as @MatkovIvan explained when introducing me to meriging from aosp). And then I created a new commit with a new name.

I'll update the name and will do force push.

eymar replied:
fixed

igordmn commented:
Make sure, that you:
- merge it as a merge commit
- don't merge as a merge commit your another PRs (GitHub remembers what you last used)
- perform empty merge `origin/integration-copy/compose-material3/dfc34b29f7e/to-jb-main` back to `integration` (not jb-main! there was a branch created by the script)

igordmn commented on compose/material3/material3-xr/api/current.txt:
```diff
@@ -0,0 +1 @@
+// Signature format: 4.0
```
As `material3-xr`/`material3-xr-adaptive` are not a part of the Material3 version group, we need to exclude them from the PR and [the script](https://github.com/JetBrains/compose-multiplatform-core/blob/jb-main/scripts/copy/copyComposeMaterial3.sh), and keep them only in `integration`.

igordmn replied:
It seems there is a duplicate `copyMaterial3.sh`. We could delete it to avoid confusion

eymar replied:
fixed

igordmn commented on compose/material3/material3/src/androidMain/kotlin/androidx/compose/material3/PrecisionPointer.android.kt:
```diff
@@ -26,31 +26,26 @@ import androidx.compose.runtime.DisposableEffect
 import androidx.compose.ui.platform.LocalContext
 import androidx.core.content.getSystemService
 
-internal actual val shouldUsePrecisionPointerComponentSizing: Boolean
```
I didn't find an issue in YouTrack for implementing this (we had to add it during the previous merges).

Please, create "Implement `shouldUsePrecisionPointerComponentSizing` setting" "Introduced in ... merge" (now it is not an expect/actual, but we still need to set it)

eymar replied:
There is this - https://youtrack.jetbrains.com/issue/CMP-9305 , I added a comment there

eymar committed:
Exclude material3-xr group from scripts/copy

Also, delete copyMaterial3.sh as it is covered by a newer pair of scripts: copyComposerMaterial3.sh and copyComposerMaterial3Adaptive.sh
